python/lsst/ip/diffim/getTemplate.py

```diff
@@ -223,89 +232,202 @@ def run(self, coaddExposures, bbox, wcs, dataIds, physical_filter=None, **kwargs
+            "patch" entries.
 
-            exp = afwImage.ExposureF(finalBBox, finalWcs)
-            exp.maskedImage.set(np.nan, afwImage.Mask.getPlaneBitMask("NO_DATA"), np.nan)
-            exp.maskedImage.assign(warped.maskedImage, warped.getBBox())
-
-            maskedImageList.append(exp.maskedImage)
-            weightList.append(1)
-            record = tractsCatalog.addNew()
-            record.setPsf(coaddExposure.getPsf())
-            record.setWcs(coaddExposure.getWcs())
-            record.setPhotoCalib(coaddExposure.getPhotoCalib())
-            record.setBBox(coaddExposure.getBBox())
-            record.setValidPolygon(afwGeom.Polygon(geom.Box2D(coaddExposure.getBBox()).getCorners()))
-            record.set(tractKey, dataId['tract'])
-            record.set(patchKey, dataId['patch'])
-            record.set(weightKey, 1.)
-            nPatchesFound += 1
-
-        if nPatchesFound == 0:
-            raise pipeBase.NoWorkFound("No patches found to overlap detector")
-
-        # Combine images from individual patches together
-        statsFlags = afwMath.stringToStatisticsProperty('MEAN')
-        statsCtrl = afwMath.StatisticsControl()
-        statsCtrl.setNanSafe(True)
-        statsCtrl.setWeighted(True)
-        statsCtrl.setCalcErrorMosaicMode(True)
-
-        templateExposure = afwImage.ExposureF(finalBBox, finalWcs)
-        templateExposure.maskedImage.set(np.nan, afwImage.Mask.getPlaneBitMask("NO_DATA"), np.nan)
-        xy0 = templateExposure.getXY0()
-        # Do not mask any values
-        templateExposure.maskedImage = afwMath.statisticsStack(maskedImageList, statsFlags, statsCtrl,
-                                                               weightList, clipped=0, maskMap=[])
-        templateExposure.maskedImage.setXY0(xy0)
+        Returns
+        -------
+        images : `list` [`lsst.afw.image.MaskedImage`]
+            MaskedImages of each of the input exposures, for warping.
+        catalog : `lsst.afw.table.ExposureCatalog`
+            Catalog of metadata for each exposure
+        totalBox : `lsst.geom.Box2I`
+            The union of the bounding boxes of all the input exposures.
+        """
+        catalog = afwTable.ExposureCatalog(self.schema)
+        catalog.reserve(len(exposures))
+        images = [exposure.maskedImage for exposure in exposures]
+        totalBox = geom.Box2I()
+        for coadd, dataId in zip(exposures, dataIds):
+            totalBox = totalBox.expandedTo(coadd.getBBox())
+            record = catalog.addNew()
+            record.setPsf(coadd.psf)
+            record.setWcs(coadd.wcs)
+            record.setPhotoCalib(coadd.photoCalib)
+            record.setBBox(coadd.getBBox())
+            record.setValidPolygon(afwGeom.Polygon(geom.Box2D(coadd.getBBox()).getCorners()))
+            record.set("tract", dataId["tract"])
+            record.set("patch", dataId["patch"])
+            # TODO: Modify this once we know which coadds were actually included?
```

Comment:
I don't know if it's necessary for anything downstream. I suppose I could remove it and see if anything breaks, since it provides no information (and at this point, actually incorrect information).

Reply:
Ah,`CoaddPsf` needs there to be a `weights` field, so I'm going to leave it as is.

Reply:
Hmm.  It probably doesn't matter much that this value is incorrect, then, since I'd expect the coadd PSFs from two different patches in an overlap region to be very similar, and downstream code is probably not super sensitive to PSF quality.  But we should update this comment to say something like that.